plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/SwitchMediatorPropertiesEditionPartForm.java

```diff
@@ -971,35 +974,46 @@ protected Composite createSourceXPathWidget(FormToolkit widgetFactory, final Com
        widgetFactory.paintBordersFor(parent);
        GridData valueData = new GridData(GridData.FILL_HORIZONTAL);
        sourceXPathText.setLayoutData(valueData);
-       sourceXPathText.addFocusListener(new FocusAdapter() {
-           /**
-            * @see org.eclipse.swt.events.FocusAdapter#focusLost(org.eclipse.swt.events.FocusEvent)
-            * 
-            */
+
+       sourceXPathText.addMouseListener(new MouseAdapter() {
+           
            @Override
-           @SuppressWarnings("synthetic-access")
-           public void focusLost(FocusEvent e) {
+           public void mouseDown( MouseEvent event ) {
+               openNamespacedPropertyEditor(parent);
            }
-
-           /**
-            * @see org.eclipse.swt.events.FocusAdapter#focusGained(org.eclipse.swt.events.FocusEvent)
-            */
+           
+       });
+       
+       sourceXPathText.addKeyListener(new KeyListener() {
+                       
            @Override
-           public void focusGained(FocusEvent e) {
-               EEFNameSpacedPropertyEditorDialog nspd = new EEFNameSpacedPropertyEditorDialog(parent.getShell(), SWT.NULL, sourceXPath);
-               //valueExpression.setPropertyValue(valueExpressionText.getText());
-               nspd.open();
-               sourceXPathText.setText(sourceXPath.getPropertyValue());
-               propertiesEditionComponent.firePropertiesChanged(new PropertiesEditionEvent(SwitchMediatorPropertiesEditionPartForm.this, EsbViewsRepository.SwitchMediator.Properties.sourceXPath, PropertiesEditionEvent.COMMIT, PropertiesEditionEvent.SET, null, getSourceXPath()));
+           public void keyPressed(KeyEvent e) {
+               openNamespacedPropertyEditor(parent);
            }
+           
+           @Override
+           public void keyReleased(KeyEvent e) {}
+           
        });
+       
        EditingUtils.setID(sourceXPathText, EsbViewsRepository.SwitchMediator.Properties.sourceXPath);
        EditingUtils.setEEFtype(sourceXPathText, "eef::Text");
        Control itemHelp = FormUtils.createHelpButton(widgetFactory, parent, propertiesEditionComponent.getHelpContent(EsbViewsRepository.SwitchMediator.Properties.sourceXPath, EsbViewsRepository.FORM_KIND), null); //$NON-NLS-1$
        sourceXPathElements = new Control[] {itemLabel, sourceXPathText, itemHelp};
        return parent;
     }
    
+    private void openNamespacedPropertyEditor(final Composite parent) {
+        EEFNameSpacedPropertyEditorDialog nspd = new EEFNameSpacedPropertyEditorDialog(parent.getShell(), SWT.NULL,
+                sourceXPath);
+        // valueExpression.setPropertyValue(valueExpressionText.getText());
```

Comment:
Remove commented line.

Reply:
Fixed in https://github.com/wso2/devstudio-tooling-esb/commit/1d53e06b88e566a6d316666c06dd48202d396d30